src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/spreadsheet/PriceSheet.kt

```diff
@@ -1,94 +1,181 @@
 package uk.gov.justice.digital.hmpps.pecs.jpc.spreadsheet
 
+import org.apache.pdfbox.util.Hex
 import org.apache.poi.ss.usermodel.BuiltinFormats
 import org.apache.poi.ss.usermodel.CellStyle
 import org.apache.poi.ss.usermodel.FillPatternType
 import org.apache.poi.ss.usermodel.HorizontalAlignment
 import org.apache.poi.ss.usermodel.IndexedColors
 import org.apache.poi.ss.usermodel.Row
 import org.apache.poi.ss.usermodel.Sheet
+import org.apache.poi.ss.util.CellRangeAddress
+import org.apache.poi.xssf.usermodel.XSSFCellStyle
+import org.apache.poi.xssf.usermodel.XSSFColor
 import uk.gov.justice.digital.hmpps.pecs.jpc.move.Journey
 import uk.gov.justice.digital.hmpps.pecs.jpc.move.Move
 import uk.gov.justice.digital.hmpps.pecs.jpc.price.Supplier
 import java.time.LocalDate
 import java.util.concurrent.atomic.AtomicInteger
+import kotlin.math.max
 
-abstract class PriceSheet(val sheet: Sheet, private val header: Header) {
+abstract class PriceSheet(
+  val sheet: Sheet,
+  private val header: Header,
+  private val subheading: String? = null,
+  private val dataColumns: List<DataColumn> = listOf()
+) {
 
-  protected val rowIndex: AtomicInteger = AtomicInteger(9)
-  protected fun createRow(rIndex: Int = rowIndex.getAndIncrement()): Row = sheet.createRow(rIndex)
-  protected fun getRow(rIndex: Int = rowIndex.getAndIncrement()): Row = sheet.getRow(rIndex)
+  enum class DataColumn(val label: String, val width: Int = 4500) {
```

Comment:
There is nothing particular smart about the width attribute.  I have simply applied the numbers having done numerous spreadsheet generations and then viewing the output.  Apache POI doesn't provide any means to do this easily and this will be just fine.